test/server/cards/02_SHD/units/GentleGiant.spec.ts

```diff
@@ -0,0 +1,60 @@
+describe('Gentle Giant', function() {
+    integration(function(contextRef) {
+        describe('Gentle Giant\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [
+                            { card: 'gentle-giant', damage: 1 },
+                            { card: 'r2d2#ignoring-protocol', damage: 3 },
+                            { card: 'c3po#protocol-droid', damage: 1 }],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'wampa', damage: 1 }]
+                    }
+                });
+            });
+
+            it('heal a target', function () {
+                const { context } = contextRef;
+
+                const reset = (passAction = true) => {
+                    context.gentleGiant.exhausted = false;
+                    context.r2d2.damage = 3;
+                    if (passAction) {
+                        context.player2.passAction();
+                    }
+                };
+
+                // CASE 1: Heal a target
+                // Attack
+                context.player1.clickCard(context.gentleGiant);
+                expect(context.gentleGiant).toBeInLocation('ground arena');
+                expect(context.player1).toBeAbleToSelectExactly([context.p2Base, context.wampa]);
+                context.player1.clickCard(context.p2Base);
+
+                // Healing Target
+                expect(context.player1).toBeAbleToSelectExactly([context.r2d2, context.c3po, context.wampa]);
+                context.player1.clickCard(context.r2d2);
+
+                // Confirm Results
+                expect(context.gentleGiant.exhausted).toBe(true);
+                expect(context.r2d2.damage).toBe(2);
+
+                reset();
+
+                // CASE 2: Should be able to pass
```

Comment:
Instead of adding this case you can just add `expect(context.player1).toHavePassAbilityButton();` at line 39 below the selection prompt. Note that the button sometimes has different names (we have a work item to fix that) so it may need to be like `toHaveChooseNoTargetsButton()` or something depending on what you see in the error message.

If the pass button appears we can safely assume it works without testing it explicitly.